spec/lib/stylesheet/manager_spec.rb

```diff
@@ -872,6 +872,67 @@ def write(x)
       content = StylesheetCache.last.content
       expect(content).to match(/# sourceMappingURL=[^\/]+\.css\.map\?__ws=test\.localhost/)
     end
+
+    it "generates precompiled CSS with a missing upload" do
+      image = file_from_fixtures("logo.png")
+      upload = UploadCreator.new(image, "logo.png").create_for(-1)
+
+      scheme = ColorScheme.create!(name: "scheme")
+      core_targets = [:desktop, :mobile, :desktop_rtl, :mobile_rtl, :admin, :wizard]
+      theme_targets = [:desktop_theme, :mobile_theme]
+
+      default_theme = Fabricate(:theme, color_scheme: scheme).tap do |t|
+        field = ThemeField.create!(
+          theme_id: t.id,
+          target_id: Theme.targets[:common],
+          name: "logo",
+          value: "",
+          upload_id: upload.id,
+          type_id: ThemeField.types[:theme_upload_var]
+        )
+
+        t.set_field(
+          target: :common,
+          name: :scss,
+          value: "body { background: url($logo); }"
+        )
+
+        t.save!
+      end
+
+      default_theme.set_default!
+      upload.delete
+      StylesheetCache.destroy_all
+
+      # only core
+      output = capture_output(:stderr) do
+        Stylesheet::Manager.precompile_css
+      end
+
+      results = StylesheetCache.pluck(:target)
+      expect(results.size).to eq(core_targets.size)
+
+      StylesheetCache.destroy_all
+
+      # only themes
+      output = capture_output(:stderr) do
+        Stylesheet::Manager.precompile_theme_css
+      end
+
+      # Ensure we force compile each theme once
+      expect(output.scan(/#{default_theme.name}/).length).to eq(7)
```

Comment:
This is failing in the run, but I don't think we need this check, it doesn't really matter what theme name is or how long it is.